### ISSUE TITLE
Delayed API response causes dropdown to show even after text has been cleared.

### ISSUE THREAD

**Issue by JulianGriggs** [ghiden/angucomplete-alt#254]
In the specific case where the options for the directive include:
```
clear-selected="true"
override-suggestions="true"
remote-api-handler="apiHandler"
selected-object="object"
pause="0"
text-no-results="false"
text-searching="false"
```
If a user types an input which triggers the API call but then quickly use the enter key or tab key to select their custom input (made possible by the override-suggestions option) the delayed API response, upon success, will cause the dropdown field to show even though the text in the input field had been cleared upon selection of the custom value (due to the clear-selected option).  

I have determined that this behavior can be fixed by changing the conditional check on line 572 from 
```
if (scope.results.length === 0 && !displayNoResults) {
          scope.showDropdown = false;
}
```
to
```
if ((scope.results.length === 0 && !displayNoResults) || (!scope.searchStr && minlength != 0)) {
          scope.showDropdown = false;
}
```
Unfortunately this solution is not perfect because in the case where minlength == 0, the delayed api response will be for whatever the custom input was, rather than for the empty input string.

**Comment by ghiden** [comment]
Thanks for reporting. I'll have a look.

**Comment by ghiden** [comment]
Try modifying handleOverrideSuggestions like this. If this works, I'll commit this.

```js
function handleOverrideSuggestions(event) {
  if (scope.overrideSuggestions &&
      !(scope.selectedObject && scope.selectedObject.originalObject === scope.searchStr)) {
    if (event) {
      event.preventDefault();
    }
    // cancel search timer
    $timeout.cancel(searchTimer);
    // cancel http request
    cancelHttpRequest();

    setInputString(scope.searchStr);
  }
}
```

**Comment by JulianGriggs** [comment]
The fix works for me.  Thank you for looking into the issue so promptly!

**Comment by ghiden** [comment]
Thanks for testing it. I just released v1.2.3.
